### ISSUE TITLE
[BUG] BED PID Autotune and PSU auto power-on

### ISSUE THREAD

**Issue by pietchaki** [MarlinFirmware/Marlin#19056]
### Bug Description

Trying to run PID AutoTune for the bed, on a printer that is using the PS_ON pin to automatically power on the main power suply, results in a failure.

### My Configurations

I'm using an ATX power supply, with it's PS-ON cable (green-wire) connected to RAMPS PS-ON pin. 
[ConfigFiles.zip](https://github.com/MarlinFirmware/Marlin/files/5085000/ConfigFiles.zip)

### Steps to Reproduce

1. Use some kind of auto power on feature on the main PSU controlled by Marlin.
2. Set marlin configurations:
```
#define PIDTEMPBED
#define PSU_CONTROL
#define PSU_ACTIVE_HIGH false
#define PSU_DEFAULT_OFF 
#define AUTO_POWER_CONTROL
```
3. Run Bed PID AutoTune: M303 C10 D1 E-1 S95 U1
4. receive error:
```
Error:Heating failed, system stopped! Heater_ID: bed
Error:Printer halted. kill() called!
```

**Expected behavior:**
Turn PSU On, then run PID tune.

**Actual behavior:** [What actually happens]
Marlin keeps PSU OFF, and try to heat hotend or bed. This results in thermal runaway, as it is trying to heat but gets no increase in temperature read.

#### Additional Information
With my current configuration (using the auto power-on feature), I can set temperatures manually, and can print without any problems. Only the PID tune fails.
When I forced the PSU to power-on, disconnecting the PS-ON cable from RAMPS and connecting it to GROUND, then PID Tune was successful.
When running PID tune, the "target temperature" on the display, Pronterface and Octoprint stays at 0 (zero). I believe the target temperature is what causes the PSU to Power ON when I set the printer to pre-heat, configurable by `"#define AUTO_POWER_E_TEMP".`

### Possible Solutions:
One approach could be setting the "target temperature" thing, as this would cause the PSU to power on.
Another way would be to, before starting the PID tune procedures, check if PSU_CONTROL is enabled and then set the PS_ON pin.

**Comment by GMagician** [comment]
Just a fast code change...if it compiles you may test it

**Comment by pietchaki** [comment]
Hi, sorry for the delay,
I just got the latest Bugfix_2.0.x, builded it and uploaded to my board.
Now when I send M303, the PSU is turned ON. BUT the heater mosfet stays OFF.
Both for the extruder and bed. The Mosfet LEDs on the RAMPS board stays off, also the one in my bed (mk2b).
Again, I can heat them by sending M104 or M140 and all works normal.

I believe I correctly merged my settings into the new file, anyway heres the one I used in this test:
[Configuration_bugfix.zip](https://github.com/MarlinFirmware/Marlin/files/5099889/Configuration_bugfix.zip)

Also, thanks for the update!

Edit:
I was trying some things here and I saw that when I powered On the PSU with an M80 command, then run the PID tune, it worked without issues.

**Comment by GMagician** [comment]
This sound really weird. Once you have power you should have the mosfet working. Code change is not "config" dependent so I doubt the problem is there.

**Comment by boelle** [comment]
@pietchaki PR was merged just 2 days ago, did you download again after it was merged?

**Comment by pietchaki** [comment]
Yes, I made sure to have code @GMagician made. Confirmed by opening the files in VS Code.
Anyway, I tried to rebuild with the newest bugfix version today, but I got the same problem.
I noticed that if the power is off and I send a M303, mosfet LED blinks one time and then stays off. This does not happen if I power the PSU via M80 command before the PID tune. Also if power is OFF and  I send an M104, the PSU powers ON and the mosfet keeps ON.

Here are 2 videos I made showing how it behaves:
1- [With power OFF, then run "M303 E0 C8 S100"](https://drive.google.com/file/d/1CCtQxbtabjim04fOgUdU8oFBoMKpXeNh/view?usp=sharing)
I turned the camera to the board before sending the command, to show that the mosfet LED just blinks.

2- [Run M80 before, to power the PSU ON, then run the same M303 as before](https://drive.google.com/file/d/1CCtQxbtabjim04fOgUdU8oFBoMKpXeNh/view?usp=sharing)

**Comment by GMagician** [comment]
@pietchaki I don't have PSU power control so I can't test it. I'll try to take a look to see if I can see something. Maybe some sort of "abort" is done

**Comment by pietchaki** [comment]
Ok, 
I was reading the files you had changed and noted that the ```Power::power_on()``` procedure calls for ```safe_delay(PSU_POWERUP_DELAY);```(power.cpp line 107), and this then calls for ```thermalManager.manage_heater()"```. In this method, ```temp_hotend[e].soft_pwm_amount``` is updated (temperature.cpp line 1055).
When I run a M303, this causes the "soft_pwm_amount" to be reset to zero after PSU powers ON. I'm not sure how to fix this for when running PID without messing with anything else...

I had put a "debug" message right before that update:
file: src/module/temperature.cpp line: 1055
```cpp
      uint8_t previous_temp = temp_hotend[e].soft_pwm_amount;
      temp_hotend[e].soft_pwm_amount = (temp_hotend[e].celsius > temp_range[e].mintemp || is_preheating(e)) && temp_hotend[e].celsius < temp_range[e].maxtemp ? (int)get_pid_output_hotend(e) >> 1 : 0;
      SERIAL_ECHOLNPAIR("--> HotEnd:", e, " hotendTarget:", temp_hotend[e].target, " old_soft_pwm:", previous_temp , " New_Soft_PWM:", temp_hotend[e].soft_pwm_amount);
```
And got this as result:
```
Recv: --> HotEnd:0 hotendTarget:0 old_soft_pwm:0 New_Soft_PWM:0 // this is the "debug" message I inserted
Recv: --> HotEnd:0 hotendTarget:0 old_soft_pwm:0 New_Soft_PWM:0
Send: M303 E0 S100 C5
Recv: PID Autotune start
Recv: --> HotEnd:0 hotendTarget:0 old_soft_pwm:127 New_Soft_PWM:0  // PID set soft_pwm_amount  to 127, power On process reset it to zero
Recv: --> HotEnd:0 hotendTarget:0 old_soft_pwm:0 New_Soft_PWM:0
Recv: Error:Heating failed, system stopped! Heater_ID: 0
```

**Comment by GMagician** [comment]
@pietchaki try new PR and let me know

**Comment by pietchaki** [comment]
Great!
It's working correctly now! 
Thank you!

**Comment by boelle** [comment]
> It's working correctly now

when the PR is merged then click close if it still works

**Comment by boelle** [comment]
@pietchaki does it still work? PR was merged

**Comment by pietchaki** [comment]
Sorry for the delay, I was testing the merged version.
All working fine here!
